src/content/docs/en/verdaccio.mdx

```diff
@@ -0,0 +1,326 @@
+  it("returns the correct value", () => {
+    expect(identity("Foo Bar")).toEqual("Foo Bar");
+  });
+});
+```
+
+Finally, let's add a script to the `package.json` so we can easily run the test suite:
+
+```json
+...
+   "scripts: {
+    ...
+    "test": "vitest run",
+    ...
+   }
+```
+
+Now you should be able to run your automated test suite with `npm run test` & see something similar to the following:
+
+<img src={convertToBaseUrl("verdaccio/verdaccio-vitest.png")} />
+
+### Testing as a CI job
+
+If you've setup CI jobs before, you may already have an idea of how to run the test suite. The tricky part ends up being standing up the verdaccio instance and getting your package published from the CI environment.
+
+Luckily, Github Actions allows you to provide a `services` definition to your jobs, which allows you to [run containerized services alongside your jobs](https://docs.github.com/en/actions/using-containerized-services).
+
+For example you may have the following in your workflow:
+
+```yaml
+jobs:
+  e2e-test:
+    services:
+      verdaccio-instance:
+        image: verdaccio/verdaccio
+        ports:
+          - 4873:4873
+```
+
+The above snippet is simply saying to run the verdaccio docker image as part of the `e2e-test` job. We forward the port that verdaccio runs on so we can access it from the job.
+
+From here, you should be able use the verdaccio instance for both installing & publishing packages from your CI job. For a full example of a job specification, see below:
+
+```yaml
+jobs:
+  e2e-test:
+    services:
+      verdaccio-instance:
+        image: verdaccio/verdaccio
+        ports:
+          - 4873:4873
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js 20
+        uses: actions/setup-node@v3
+        with:
+          node-version: 20
+          cache: "npm"
+        # Setup verdaccio as the default npm registry
+      - run: echo "registry=http://localhost:4873" > ~/.npmrc
+      - name: Publish Package
+        run: |
+          {
+            cd verdaccio-test
+            npm version 1.0.0-test-ci --git-tag-version=false
+            npm ci
+            npx npm-cli-login -u test -p 1234 -e test@domain.test -r http://localhost:4873
+            npm publish --registry http://localhost:4873
+          }
+      - name: "Test published package"
+        run: |
+          {
+            cd verdaccio-test-install
+            npm install
+            npm install my-test-pkg1@1.0.0-test-ci
+            npm run test
+          }
+```
+
+## Closing Thoughts
+
+While this setup may not be necessary for every project, I think that Verdaccio provides a superior service to alternatives like `npm link` & `yalc`, due to simulateneously being a full-fledged npm registry as well and providing an ephemeral environment that can be recreated simply with a docker image. I hope this article was useful for you in getting started with e2e testing npm packages. Happy Coding!
```

Comment:
```suggestion

Before implementing this for every one of your projects, take a while to think about whether you actually need it. Verdaccio is a very helpful tool, but it admittedly adds complexity that's not necessary for every project. Have you run into problems that can't be debugged without first having the package published? If you aren't, it might not be time to adopt Verdaccio yet.

While this setup may not be necessary for every project, I think that Verdaccio provides a superior service to alternatives like `npm link` & `yalc`, due to simulateneously being a full-fledged npm registry as well and providing an ephemeral environment that can be recreated simply with a docker image. I hope this article was useful for you in getting started with e2e testing npm packages. Happy Coding!
```